### PR TITLE
feat: Do not require test tools or UCD Processes for 2014 listings that have ICS

### PR DESCRIPTION
**Pull request by kekey1**
OCD-3131

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2014/TestTool2014Reviewer.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Autowired
+    public TestTool2014Reviewer(final ErrorMessageUtil msgUtil) {
+        this.msgUtil = msgUtil;
+    }
+
+    @Override
+    public void review(final CertifiedProductSearchDetails listing) {
+        for (CertificationResult cert : listing.getCertificationResults()) {
+            if (cert.isSuccess()) {
+                if (cert.getTestToolsUsed() != null && cert.getTestToolsUsed().size() > 0) {
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2014/TestTool2014Reviewer.java]: Could this and the previous line be combined? To reduce the nesting?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2014/TestTool2014Reviewer.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                    Iterator<CertificationResultTestTool> testToolIter = cert.getTestToolsUsed().iterator();
+                    while (testToolIter.hasNext()) {
+                        CertificationResultTestTool testTool = testToolIter.next();
+                        if (!StringUtils.isEmpty(testTool.getTestToolName())) {
+                            // require test tool version if not ICS
+                            if (StringUtils.isEmpty(testTool.getTestToolVersion())) {
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2014/TestTool2014Reviewer.java]: Similar question regarding nesting

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2014/TestTool2014Reviewer.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                                if (listing.getIcs() != null && listing.getIcs().getInherits() != null
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2014/TestTool2014Reviewer.java]: Maybe similar question; this one has an else, so maybe it's more clear on its own line

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/TestTool2015Reviewer.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    @Autowired
+    public TestTool2015Reviewer(final ErrorMessageUtil msgUtil) {
+        this.msgUtil = msgUtil;
+    }
+
+    @Override
+    public void review(final CertifiedProductSearchDetails listing) {
+        for (CertificationResult cert : listing.getCertificationResults()) {
+            if (cert.isSuccess()) {
+                if (cert.getTestToolsUsed() != null && cert.getTestToolsUsed().size() > 0) {
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/TestTool2015Reviewer.java]: Same nesting questions

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/pendingListing/reviewer/edition2014/TestTool2014Reviewer.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Autowired
+    public TestTool2014Reviewer(final ErrorMessageUtil msgUtil) {
+        this.msgUtil = msgUtil;
+    }
+
+    public void review(final PendingCertifiedProductDTO listing) {
+        for (PendingCertificationResultDTO cert : listing.getCertificationCriterion()) {
+            if (cert.getMeetsCriteria() != null && cert.getMeetsCriteria()) {
+                if (cert.getTestTools() != null && cert.getTestTools().size() > 0) {
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/pendingListing/reviewer/edition2014/TestTool2014Reviewer.java]: Nesting question

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/pendingListing/reviewer/edition2015/TestTool2015Reviewer.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    private ErrorMessageUtil msgUtil;
+
+    @Autowired
+    public TestTool2015Reviewer(final ErrorMessageUtil msgUtil) {
+        this.msgUtil = msgUtil;
+    }
+
+    public void review(final PendingCertifiedProductDTO listing) {
+        for (PendingCertificationResultDTO cert : listing.getCertificationCriterion()) {
+            if (cert.getMeetsCriteria() != null && cert.getMeetsCriteria()) {
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/pendingListing/reviewer/edition2015/TestTool2015Reviewer.java]: I want a "nest" emoji; can't find one...

**Review by andlar — APPROVED** [review]

File: chpl/chpl-api/src/test/java/gov/healthit/chpl/CHPLTestConfig.java [hunk 7]
```diff
@@ -69,8 +70,10 @@
 @Configuration
 @Import({ChplTestCacheConfig.class})
 @EnableGlobalMethodSecurity(prePostEnabled = true)
-@PropertySource("classpath:/environment.test.properties")
-@EnableCaching
+@PropertySources({
+    @PropertySource("classpath:/environment.test.properties"),
+    @PropertySource("classpath:/lookup.test.properties"),
+})@EnableCaching
```
Comment (Plow74) [inline chpl/chpl-api/src/test/java/gov/healthit/chpl/CHPLTestConfig.java]: Looks like this annotation is sneaking in behind the one above it. Nothing major, just looks strange and could be missed if looking over the class quickly.

**Review by Plow74 — APPROVED** [review]

**Pull request merged**